src/lib/ota/jethome.ts

```diff
@@ -0,0 +1,109 @@
+const baseurl = 'https://fw.jethome.ru';
+const deviceurl = baseurl + '/api/devices/';
+import * as common from './common';
+import {Logger, Zh, Ota, KeyValueAny} from '../types';
+const axios = common.getAxios();
+import fs from 'fs';
+import * as URI from 'uri-js';
+import path from 'path';
+
+let overrideIndexFileName: string = null;
+let dataDir: string = null;
+
+/**
+ * Helper functions
+ */
+
+
+function isValidUrl(url: string) {
+    let parsed;
+    try {
+        parsed = URI.parse(url);
+    } catch (_) {
+        return false;
+    }
+    return parsed.scheme === 'http' || parsed.scheme === 'https';
+}
+
+function readLocalFile(fileName: string, logger: Logger) {
```

Comment:
This and also the `isValidUrl` are the same as the ones as in `zigbeeOTA.ts`, please call those function instead of copying them here.

Reply:
Fixed. isValidUrl moved to common.ts